tests/sceneManager_test.cpp

```diff
@@ -39,22 +40,73 @@ TEST_F(SceneManagerTest, EnergyCollectorPositionsCheck)
+        << " point from: " << parallelXAxis;
 
-    Ray parallelYaxis(core::Vec3(0, 0, 0), core::Vec3(0, 1, 0));
-    ASSERT_TRUE(performHitAtEnergyCollectors(parallelYaxis, &hitData)) << "no hit: " << parallelYaxis;
+    Ray parallelYaxis(kVZero, core::Vec3(0, 1, 0));
+    ASSERT_TRUE(performHitAtEnergyCollectors(parallelYaxis, &hitData))
+        << "no hit: " << parallelYaxis;
 
     Vec3 referenceCollisionPoint3(0, manager.simulatiorRadius() / 2 - manager.collectorRadius(), 0);
-    ASSERT_EQ(hitData.collisionPoint, referenceCollisionPoint3) << " point from: " << parallelYaxis;
+    ASSERT_EQ(hitData.collisionPoint, referenceCollisionPoint3)
+        << " point from: " << parallelYaxis;
+
+    Ray atSixtyDegreesAlongX(kVZero, core::Vec3(std::cos(constants::kPi / 3), 0,
+                                                std::sin(constants::kPi / 3)));
+    ASSERT_TRUE(performHitAtEnergyCollectors(atSixtyDegreesAlongX, &hitData))
+        << "no hit: " << atSixtyDegreesAlongX;
+
+    Vec3 referenceCollisionAtSixty(
+        manager.simulatiorRadius() / 2 * std::cos(constants::kPi / 3) -
+            manager.collectorRadius() * std::cos(constants::kPi / 3),
+        0,
+        manager.simulatiorRadius() / 2 * std::sin(constants::kPi / 3) -
+            manager.collectorRadius() * std::sin(constants::kPi / 3));
+
+    ASSERT_EQ(hitData.collisionPoint, referenceCollisionAtSixty)
+        << "Invalid hit from " << atSixtyDegreesAlongX
+        << " to: " << referenceCollisionAtSixty;
+
+    Ray atThirtyDegreesAlongX(kVZero,
+                              core::Vec3(std::cos(constants::kPi / 6), 0,
+                                         std::sin(constants::kPi / 6)));
+    ASSERT_TRUE(performHitAtEnergyCollectors(atThirtyDegreesAlongX, &hitData))
+        << "no hit: " << atThirtyDegreesAlongX;
+
+    Vec3 referenceCollisionAtThirty(
+        manager.simulatiorRadius() / 2 * std::cos(constants::kPi / 6) -
+            manager.collectorRadius() * std::cos(constants::kPi / 6),
+        0,
+        manager.simulatiorRadius() / 2 * std::sin(constants::kPi / 6) -
+            manager.collectorRadius() * std::sin(constants::kPi / 6));
+
+    ASSERT_EQ(hitData.collisionPoint, referenceCollisionAtThirty)
+        << "Invalid hit from " << atThirtyDegreesAlongX
+        << " to: " << referenceCollisionAtThirty;
+
+    Ray straightDown(kVZero, core::Vec3(0, 0, -1));
+    ASSERT_FALSE(performHitAtEnergyCollectors(straightDown, &hitData))
+        << "hit registered in: " << hitData.collisionPoint
+        << "from ray: " << straightDown;
+
+    // Ray at the edge of EnergyCollector at: Vec3(0, 0, 2)
+    Ray straightUpAbove(
+        core::Vec3(0, 0,
+                   manager.simulatiorRadius() / 2 + manager.collectorRadius()),
+        core::Vec3(0, 0, 1));
+    ASSERT_FALSE(performHitAtEnergyCollectors(straightUpAbove, &hitData))
+        << "hit registered in: " << hitData.collisionPoint
+        << "from ray: " << straightUpAbove;
```

Comment:
There is still one more non-trivial case missing.

Reply:
I added hit from inside energyCollector

Reply:
I do not see it, but sounds like a good case.

Besides this is not what I meant, so add more non-trivial cases ;-)

Reply:
I cant come up with the idea of what you mean then ;/

Reply:
I added a comment what case I mean below. The new one you added is good.

Closing this thread.